src/test/java/pl/psnc/dei/service/TranscriptionPlatformServiceTest.java

```diff
@@ -32,155 +34,155 @@
-//                .willReturn(aResponse()
-//                        .withStatus(500)
-//                        .withHeader("Content-Type", "application/json")
-//                        .withFixedDelay(2000)
-//                        .withBody("[{\"transcription\":\"test123\",\"target\":\"test321\",\"Timestamp\":\"Apr 22, 2019 12:50:57 PM\"},{\"transcription\":\"test12345\",\"target\":\"test54321\",\"Timestamp\":\"Apr 22, 2019 12:50:57 PM\"}]")));
-//        //
-//        Record testRecord = new Record();
-//        testRecord.setIdentifier("123");
-//        transcriptionPlatformService.fetchTranscriptionsFor(testRecord);
-//
-//        Assert.fail();
-//    }
-//
-//    @Test
-//    public void shouldSendAnnotationUrl() throws TranscriptionPlatformException {
-//        wireMockRule.resetAll();
-//        wireMockRule.stubFor(post(urlEqualTo("/transcription/sampleIdentifierFromTP"))
-//                .willReturn(aResponse()
-//                        .withStatus(201)
-//                        .withFixedDelay(2000)));
-//        //
-//        Transcription testTranscription = new Transcription();
-//        testTranscription.setTp_id("sampleIdentifierFromTP");
-//        testTranscription.setAnnotationId("sampleAnnotationId");
-//        transcriptionPlatformService.sendAnnotationUrl(testTranscription);
-//    }
-//
-//    @Test(expected = TranscriptionPlatformException.class)
-//    public void shouldFailOnTimeoutWhileSendingAnnotationUrl() {
-//        wireMockRule.resetAll();
-//        wireMockRule.stubFor(post(urlEqualTo("/transcription/sampleIdentifierFromTP"))
-//                .willReturn(aResponse()
-//                        .withStatus(200)
-//                        .withFixedDelay(7000)));
-//
-//        //
-//        Transcription testTranscription = new Transcription();
-//        testTranscription.setTp_id("sampleIdentifierFromTP");
-//        testTranscription.setAnnotationId("sampleAnnotationId");
-//        transcriptionPlatformService.sendAnnotationUrl(testTranscription);
-//
-//        Assert.fail();
-//    }
-//
-//    @Test(expected = TranscriptionPlatformException.class)
-//    public void shouldFailOnServerErrorWhileWhileSendingAnnotationUrl() {
-//        wireMockRule.resetAll();
-//        wireMockRule.stubFor(post(urlEqualTo("/transcription/sampleIdentifierFromTP"))
-//                .willReturn(aResponse()
-//                        .withStatus(500)
-//                        .withFixedDelay(1000)));
-//        //
-//        Transcription testTranscription = new Transcription();
-//        testTranscription.setTp_id("sampleIdentifierFromTP");
-//        testTranscription.setAnnotationId("sampleAnnotationId");
-//        transcriptionPlatformService.sendAnnotationUrl(testTranscription);
-//
-//        Assert.fail();
-//    }
+    @Ignore("Large read timeout, should be uncomment after performance improvement in tp api and timeout decreased")
+    @Test(expected = TranscriptionPlatformException.class)
```

Comment:
Actually why does the timeout here matter? Shouldn't we have a test that is independent from any external server?

Reply:
yes, but we mock our api responses and we can get this error (which we will check)  only after timeout